tests/Functional/src/com/microsoft/aad/adal/test/AuthenticationContextTest.java

```diff
@@ -984,14 +984,14 @@ public void testRefreshTokenWebRequestHasError() throws InterruptedException,
         webrequest.setReturnResponse(new HttpWebResponse(500, null, null));
         ReflectionUtils.setFieldValue(context, "mWebRequest", webrequest);
         final TestLogResponse response = new TestLogResponse();
-        response.listenLogForMessageSegments(signal, "Refresh token did not return accesstoken");
+        response.listenLogForMessageSegments(signal, "Server error message");
```

Comment:
this function unlocks signal before acquireTokenSilentAsync exits...
you should wait until acquireTokenSilentAsync  invokes callback,
and check that you get exception in result